vscode-dotnet-runtime-library/src/Acquisition/GlobalInstallerResolver.ts

```diff
@@ -0,0 +1,351 @@
+                break;
+            }
+            default:
+            {
+                const osErr = new DotnetUnexpectedInstallerOSError(new Error(`The OS ${operatingSys} is currently unsupported or unknown.`));
+                this.eventStream.post(osErr);
+                throw osErr;
+            }
+        }
+
+        switch(operatingArch)
+        {
+            case 'x64': {
+                convertedArch = operatingArch;
+                break;
+            }
+            case 'ia32': {
+                convertedArch = 'x86';
+                break;
+            }
+            case 'arm': {
+                convertedArch = operatingArch;
+                break;
+            }
+            case 'arm64': {
+                convertedArch = operatingArch;
+                break;
+            }
+            default:
+            {
+                const archErr = new DotnetUnexpectedInstallerOSError(new Error(`The architecture ${operatingArch} is currently unsupported or unknown.`));
+                this.eventStream.post(archErr);
+                throw archErr;
+            }
+        }
+
+        const desiredRidPackage = `${convertedOs}-${convertedArch}`;
+
+        const indexJson =  await this.fetchJsonObjectFromUrl(indexUrl);
+        const releases = indexJson[this.releasesJsonKey];
+        if(releases.length === 0)
+        {
+            const jsonErr = new DotnetInvalidReleasesJSONError(new Error(`${this.releasesJsonErrorString}${indexUrl}`));
+            this.eventStream.post(jsonErr);
+            throw jsonErr;
+        }
+
+        const sdks: any[] = [];
+        const releasesKeyAlias = this.releasesSdksKey; // the forEach creates a separate 'this', so we introduce this copy to reduce ambiguity to the compiler
+
+        releases.forEach(function (release : any) {
+            sdks.push.apply(sdks, release[releasesKeyAlias]);
+        });
+
+        for (const sdk of sdks)
+        {
+            const thisSDKVersion : string = sdk[this.releasesSdkVersionKey];
+            if(thisSDKVersion === specificVersion) // NOTE that this will not catch things like -preview or build number suffixed versions.
+            {
+               const thisSDKFiles = sdk[this.releasesSdkFileKey];
+               for (const installer of thisSDKFiles)
+               {
+                    if(installer[this.releasesSdkRidKey] === desiredRidPackage && this.installerMatchesDesiredFileExtension(installer, convertedOs))
+                    {
+                        const installerUrl = installer[this.releasesUrlKey];
+                        if(installerUrl === undefined)
+                        {
+                            const releaseJsonErr = new DotnetInvalidReleasesJSONError(new Error(`URL for ${desiredRidPackage} on ${specificVersion} is unavailable:
+                                The version may be Out of Support, or the releases json format used by ${indexUrl} may be invalid and the extension needs to be updated.`));
```

Comment:
Do errors like this need localization?

Reply:
VSCode itself uses this for localization. https://github.com/microsoft/vscode-l10n

Reply:
Thanks, that's a really good point. I definitely dont want to add that to this PR and will do that separately. I made an issue for it: https://github.com/dotnet/vscode-dotnet-runtime/issues/1130